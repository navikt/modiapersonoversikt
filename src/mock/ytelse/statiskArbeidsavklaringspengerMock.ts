export const statiskArbeidsavklaringspengerMock = {
    barnMedStonad: 2,
    barnetillegg: 1500,
    beregningsgrunnlag: 30000,
    dagsats: 500,
    kildesystem: 'KELVIN',
    periode: {
        fraOgMedDato: '2025-01-01',
        tilOgMedDato: '2025-12-31'
    },
    rettighetsType: 'AAP',
    saksnummer: '2023123456',
    status: 'LÃ˜PENDE',
    vedtakId: 'vedtak-001',
    vedtaksdato: '2023-02-01',
    opphorsAarsak: 'FERDIGBEHANDLET'
};
