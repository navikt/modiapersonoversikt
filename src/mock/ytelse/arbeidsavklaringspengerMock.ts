import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskArbeidsavklaringspengerMock } from 'src/mock/ytelse/statiskArbeidsavklaringspengerMock';
import type {
    Arbeidsavklaringspenger,
    ArbeidsavklaringspengerResponse
} from 'src/models/ytelse/arbeidsavklaringspenger';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockArbeidsavklaringspengerResponse(fnr: string): ArbeidsavklaringspengerResponse {
    if (fnr === aremark.personIdent) {
        return [statiskArbeidsavklaringspengerMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}arbeidsavklaringspenger`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return [];
    }

    return fyllRandomListe<Arbeidsavklaringspenger>(() => getMockArbeidsavklaringspenger(fnr), 3);
}

function getMockArbeidsavklaringspenger(fnr: string): Arbeidsavklaringspenger {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        barnMedStonad: 2,
        barnetillegg: 1500,
        beregningsgrunnlag: 30000,
        dagsats: 500,
        kildesystem: 'Arena',
        periode: {
            fraOgMedDato: fomDato,
            tilOgMedDato: tomDato
        },
        rettighetsType: 'BISTANDSBEHOV',
        saksnummer: '2023123456',
        status: 'LÃ˜PENDE',
        vedtakId: faker.number.int(100).toString(),
        vedtaksdato: '2023-02-01',
        opphorsAarsak: 'FERDIGBEHANDLET'
    };
}
