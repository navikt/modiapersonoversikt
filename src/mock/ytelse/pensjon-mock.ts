import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskPensjonMock } from 'src/mock/ytelse/statiskPensjonMock';
import type { Pensjon, PensjonResource } from 'src/models/ytelse/pensjon';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockPensjon(fnr: string): PensjonResource {
    if (fnr === aremark.personIdent) {
        return [statiskPensjonMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return null;
    }

    return fyllRandomListe<Pensjon>(() => getMockPensjonYtelser(fnr), 3);
}

function getMockPensjonYtelser(fnr: string): Pensjon {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        sakid: faker.string.alphanumeric(8),
        fomDato,
        tomDato,
        enhetId: '0129',
        sakType: 'AFP',
        sakStatus: 'OPPRETTET'
    };
}
