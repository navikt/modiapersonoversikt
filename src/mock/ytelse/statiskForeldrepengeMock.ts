import { Fødsel } from '../../models/ytelse/foreldrepenger';

export const statiskKommendeUtbetaling = {
    vedtak: { fra: '2018-01-14', til: '2017-07-15' },
    utbetalingsgrad: 79,
    utbetalingsdato: '2018-06-25',
    bruttobeløp: 466,
    arbeidsgiverNavn: 'Vedvik - Lunde',
    arbeidsgiverOrgNr: '1234567890',
    arbeidsgiverKontonr: '549801574',
    dagsats: 705,
    saksbehandler: 'Mari Johannessen',
    type: 'KONTOØVERFØRING'
};

export const statiskForeldrepengeMock: Fødsel = {
    forelder: '10108000398',
    andreForeldersFnr: '03063850003',
    antallBarn: 5,
    barnetsFødselsdato: '2019-02-11',
    dekningsgrad: 34,
    fedrekvoteTom: '2019-02-10',
    mødrekvoteTom: null,
    foreldrepengetype: 'Fødselspenger',
    graderingsdager: 55,
    restDager: 41,
    rettighetFom: '2017-11-21',
    eldsteIdDato: null,
    foreldreAvSammeKjønn: null,
    periode: [
        {
            fødselsnummer: '10108000398',
            harAleneomsorgFar: null,
            harAleneomsorgMor: null,
            arbeidsprosentMor: 58,
            avslagsårsak: 'Avslag',
            avslått: 'Avslått',
            disponibelGradering: 9,
            erFedrekvote: null,
            erMødrekvote: null,
            forskyvelsesårsak1: null,
            forskyvelsesperiode1: { fra: '2014-10-08T14:22:55.629Z', til: '2014-08-22T10:50:40.296Z' },
            forskyvelsesårsak2: null,
            forskyvelsesperiode2: { fra: '2014-12-09T16:29:21.344Z', til: '2014-08-26T10:23:03.605Z' },
            foreldrepengerFom: '2017-04-29',
            midlertidigStansDato: '2019-02-10',
            morSituasjon: 'nobis sit quo et quis',
            rettTilFedrekvote: 'Rett til fedrekvote',
            rettTilMødrekvote: 'Ingen rett til mødrekvote',
            stansårsak: 'Avsluttet',
            kommendeUtbetalinger: [statiskKommendeUtbetaling]
        }
    ],
    slutt: null,
    arbeidsforhold: [
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        },
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        }
    ],
    erArbeidsgiverperiode: null,
    arbeidskategori: 'Arbeidstaker',
    termin: '2019-02-11'
};
