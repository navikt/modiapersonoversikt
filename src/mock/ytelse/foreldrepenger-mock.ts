import faker from 'faker/locale/nb_NO';
import moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Foreldrepengerperiode,
    ForeldrepengerResponse,
    Foreldrepengerettighet
} from '../../models/ytelse/foreldrepenger';
import { getPeriode } from '../person/periodeMock';
import { fyllRandomListe, vektetSjanse } from '../utils/mock-utils';
import { getKommendeUtbetaling } from './ytelse-utbetalinger-mock';
import { KommendeUtbetaling } from '../../models/ytelse/ytelse-utbetalinger';
import { Arbeidsforhold } from '../../models/ytelse/arbeidsforhold';
import { aremark } from '../person/aremark';
import { statiskForeldrepengeMock } from './statiskForeldrepengeMock';
import { backendDatoformat } from '../../utils/dateUtils';

export function getMockForeldrepenger(fødselsnummer: string): ForeldrepengerResponse {
    if (fødselsnummer === aremark.fødselsnummer) {
        return {
            foreldrepenger: [statiskForeldrepengeMock]
        };
    }

    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'foreldrepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            foreldrepenger: null
        };
    }

    return {
        foreldrepenger: fyllRandomListe<Foreldrepengerettighet>(() => getForeldrepengerettighetMock(fødselsnummer), 2)
    };
}

export function getForeldrepengerettighetMock(fødselsnummer: string, seed?: number): Foreldrepengerettighet {
    if (seed) {
        faker.seed(Number(seed));
        navfaker.seed(seed.toString());
    }
    const erFødsel = vektetSjanse(faker, 0.5) ? true : false;
    return {
        forelder: fødselsnummer,
        andreForeldersFnr: navfaker.personIdentifikator.fødselsnummer(),
        antallBarn: navfaker.random.integer(5),
        barnetsFødselsdato: moment(faker.date.recent()).format(backendDatoformat),
        dekningsgrad: navfaker.random.integer(95),
        fedrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        mødrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        foreldrepengetype: foreldrePengeType(),
        graderingsdager: navfaker.random.integer(100),
        restDager: navfaker.random.integer(50),
        rettighetFom: vektetSjanse(faker, 0.5) ? moment(faker.date.past(2)).format(backendDatoformat) : null,
        eldsteIdDato: vektetSjanse(faker, 0.5) ? moment(faker.date.past(2)).format(backendDatoformat) : null,
        foreldreAvSammeKjønn: vektetSjanse(faker, 0.5) ? 'Begge er pappaer' : null,
        periode: fyllRandomListe<Foreldrepengerperiode>(() => getForeldrepengerperiodeMock(fødselsnummer), 5),
        slutt: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        arbeidsforhold: fyllRandomListe<Arbeidsforhold>(() => getArbeidsforholdMock(fødselsnummer), 5),
        erArbeidsgiverperiode: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidskategori: vektetSjanse(faker, 0.5) ? 'Arbeidstaker' : null,
        omsorgsovertakelse: !erFødsel ? moment(faker.date.past(2)).format(backendDatoformat) : undefined,
        termin: erFødsel ? moment(faker.date.recent()).format(backendDatoformat) : undefined
    } as Foreldrepengerettighet;
}

export function getForeldrepengerperiodeMock(fødselsnummer: string): Foreldrepengerperiode {
    return {
        fødselsnummer: fødselsnummer,
        harAleneomsorgFar: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        harAleneomsorgMor: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidsprosentMor: vektetSjanse(faker, 0.5) ? navfaker.random.integer(100) : null,
        avslagsårsak: vektetSjanse(faker, 0.5) ? 'Avslag' : null,
        avslått: vektetSjanse(faker, 0.5) ? 'Avslått' : null,
        disponibelGradering: vektetSjanse(faker, 0.5) ? navfaker.random.integer(75) : null,
        erFedrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        erMødrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        forskyvelsesårsak1: vektetSjanse(faker, 0.5) ? 'FÅRSAK1' : null,
        forskyvelsesperiode1: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        forskyvelsesårsak2: vektetSjanse(faker, 0.5) ? 'FÅRSAK2' : null,
        forskyvelsesperiode2: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        foreldrepengerFom: moment(faker.date.past(5)).format(backendDatoformat),
        midlertidigStansDato: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        morSituasjon: vektetSjanse(faker, 0.5) ? faker.lorem.words(5) : null,
        rettTilFedrekvote: vektetSjanse(faker, 0.5) ? 'Rett til fedrekvote' : 'Ingen rett til fedrekvote',
        rettTilMødrekvote: vektetSjanse(faker, 0.5) ? 'Rett til mødrekvote' : 'Ingen rett til mødrekvote',
        stansårsak: vektetSjanse(faker, 0.5) ? 'Avsluttet' : null,
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 3, true)
    };
}

function getArbeidsforholdMock(fødselsnummer: string): Arbeidsforhold {
    return {
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverKontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: vektetSjanse(faker, 0.5) ? 'Månedlig' : 'Årlig',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        sykepengerFom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonstype: 'Ikke refusjon'
    };
}

function foreldrePengeType(): string {
    return navfaker.random.arrayElement(['Fødselspenger', 'Svangerskapspenger', 'Engangsstønad']);
}
