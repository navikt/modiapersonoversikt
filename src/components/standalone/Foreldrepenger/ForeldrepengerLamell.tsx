import * as React from 'react';
import ErrorBoundary from '../../ErrorBoundary';
import ForeldrepengerLaster from './ForeldrepengerLaster';
import styled from 'styled-components/macro';
import theme from '../../../styles/personOversiktTheme';
import SetFnrIRedux from '../../../app/PersonOppslagHandler/SetFnrIRedux';

interface Props {
    fødselsnummer: string;
}
const Styles = styled.div`
    overflow-y: auto;
    .visually-hidden {
        ${theme.visuallyHidden}
    }
`;

class ForeldrepengerLamell extends React.PureComponent<Props> {
    render() {
        return (
            <ErrorBoundary boundaryName="ForeldrepengeLamell">
                <Styles>
                    <SetFnrIRedux fødselsnummer={this.props.fødselsnummer} />
                    <ForeldrepengerLaster fødselsnummer={this.props.fødselsnummer} />
                </Styles>
            </ErrorBoundary>
        );
    }
}

export default ForeldrepengerLamell;
