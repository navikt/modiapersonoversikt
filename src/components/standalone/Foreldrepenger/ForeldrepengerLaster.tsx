import * as React from 'react';
import AlertStripe, { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { Foreldrepengerettighet, ForeldrepengerResponse } from '../../../models/ytelse/foreldrepenger';
import FillCenterAndFadeIn from '../../FillCenterAndFadeIn';
import Foreldrepenger from '../../../app/personside/infotabs/ytelser/foreldrepenger/ForeldrePenger';
import RestResourceConsumer from '../../../rest/consumer/RestResourceConsumer';
import { BigCenteredLazySpinner } from '../../BigCenteredLazySpinner';
import Panel from 'nav-frontend-paneler';
import styled from 'styled-components/macro';

const StyledPanel = styled(Panel)`
    padding: 0rem;
`;
interface Props {
    fødselsnummer: string;
}
const onError = (
    <FillCenterAndFadeIn>
        <AlertStripe type="advarsel">Beklager. Det skjedde en feil ved lasting av Foreldrepenger.</AlertStripe>
    </FillCenterAndFadeIn>
);

class ForeldrepengerLaster extends React.PureComponent<Props> {
    getAktuellForeldrepengeRettighet(ForeldrepengeRettighet: Foreldrepengerettighet[] | null) {
        if (!ForeldrepengeRettighet) {
            return <AlertStripeInfo>Kunne ikke finne noen Foreldrepengerettigheter for bruker</AlertStripeInfo>;
        }

        const aktuellRettighet = ForeldrepengeRettighet.find(
            rettighet => rettighet.forelder === this.props.fødselsnummer
        );

        if (!aktuellRettighet) {
            return <AlertStripeInfo>Kunne ikke finne Foreldrepengerettighet</AlertStripeInfo>;
        }

        return (
            <StyledPanel>
                <Foreldrepenger foreldrepenger={aktuellRettighet} />
            </StyledPanel>
        );
    }

    render() {
        return (
            <RestResourceConsumer<ForeldrepengerResponse>
                getResource={restResources => restResources.foreldrepenger}
                returnOnPending={BigCenteredLazySpinner}
                returnOnError={onError}
            >
                {data => this.getAktuellForeldrepengeRettighet(data.foreldrepenger)}
            </RestResourceConsumer>
        );
    }
}

export default ForeldrepengerLaster;
