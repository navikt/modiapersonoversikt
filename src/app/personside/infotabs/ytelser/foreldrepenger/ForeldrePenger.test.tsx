import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Foreldrepenger from './ForeldrePenger';
import { statiskForeldrepengeMock } from '../../../../../mock/ytelse/statiskForeldrepengeMock';
import TestProvider from '../../../../../test/Testprovider';

test('Foreldrepengeperiode matcher snapshot', () => {
    const testRettighet = statiskForeldrepengeMock;
    const result = renderer.create(
        <TestProvider>
            <Foreldrepenger foreldrepenger={testRettighet} />
        </TestProvider>
    );

    expect(result).toMatchSnapshot();
});
