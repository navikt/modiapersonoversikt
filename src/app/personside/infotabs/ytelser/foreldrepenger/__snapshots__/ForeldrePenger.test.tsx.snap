// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foreldrepengeperiode matcher snapshot 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 dt {
  color: #645f5a;
}

.c4 > div {
  margin: 0.5625rem 0;
  padding-right: 0.7rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c12 {
  overflow: auto;
  box-shadow: 0 0 0 0.07rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c12 [role='table'] {
  text-align: left;
  width: 100%;
}

.c12 [role='table'] thead,
.c12 [role='table'] .thead[role='rowgroup'] {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: 600;
}

.c12 [role='table'] [role='cell'],
.c12 [role='table'] [role='columnheader'] {
  padding: 0.7rem;
  vertical-align: middle;
}

.c12 [role='table'] [role='columnheader'] {
  padding-bottom: 0.7rem;
}

.c12 [role='table'] [role='row']:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c9 {
  background-color: #cce1f3;
  padding: 0.2rem 0.9375rem;
}

.c8 ol {
  padding: 0;
  margin: 0;
}

.c11 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  white-space: nowrap;
}

.c11 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c3 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c3 > h3 {
  margin-bottom: 1rem;
}

.c10 {
  padding: .5rem;
}

.c6 span:not(:last-child):after {
  content: '.';
}

.c5 li:not(:first-child) {
  padding-top: 1rem;
  margin-top: 1rem;
  border-top: solid 0.0625rem #b7b1a9;
}

.c7 {
  margin-top: 1rem;
}

.c1 {
  padding: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 > * {
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 {
  padding: 0rem;
}

<article>
  <div
    aria-labelledby="Helt tilfeldig ID"
    className="panel c0"
  >
    <h2
      className="sr-only"
      id="Helt tilfeldig ID"
      tabIndex={-1}
    >
      Foreldrepengerettighet
    </h2>
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <article
          aria-labelledby="Helt tilfeldig ID"
          className="c3"
        >
          <h3
            className="typo-ingress"
            id="Helt tilfeldig ID"
          >
            Om foreldrepengeretten
          </h3>
          <dl
            className="c4"
          >
            <div>
              <dt
                className="typo-normal"
              >
                Foreldrepengetype
              </dt>
              <dd
                className="typo-element"
              >
                Fødselspenger
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Dekningsgrad
              </dt>
              <dd
                className="typo-element"
              >
                34%
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Rettighet fra dato
              </dt>
              <dd
                className="typo-element"
              >
                29.04.2017
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Graderingsdager
              </dt>
              <dd
                className="typo-element"
              >
                55
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Restdager
              </dt>
              <dd
                className="typo-element"
              >
                41
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Maksdato
              </dt>
              <dd
                className="typo-element"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Arbeidskategori
              </dt>
              <dd
                className="typo-element"
              >
                Arbeidstaker
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Fedrekvote til og med
              </dt>
              <dd
                className="typo-element"
              >
                10.02.2019
              </dd>
            </div>
          </dl>
        </article>
        <article
          aria-labelledby="Helt tilfeldig ID"
          className="c3"
        >
          <h3
            className="typo-ingress"
            id="Helt tilfeldig ID"
          >
            Om barnet
          </h3>
          <dl
            className="c4"
          >
            <div>
              <dt
                className="typo-normal"
              >
                Termindato
              </dt>
              <dd
                className="typo-element"
              >
                11.02.2019
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Fødselsdato
              </dt>
              <dd
                className="typo-element"
              >
                11.02.2019
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Annen forelder
              </dt>
              <dd
                className="typo-element"
              >
                03063850003
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Antall barn
              </dt>
              <dd
                className="typo-element"
              >
                5
              </dd>
            </div>
          </dl>
        </article>
      </div>
      <article
        aria-labelledby="Helt tilfeldig ID"
        className="c3"
      >
        <h3
          className="typo-ingress"
          id="Helt tilfeldig ID"
        >
          Arbeidssituasjon
        </h3>
        <ol
          aria-label="Arbeidsgivere"
          className="c5"
        >
          <li>
            <dl
              className="c4"
            >
              <div>
                <dt
                  className="typo-normal"
                >
                  Arbeidsgiver
                </dt>
                <dd
                  className="typo-element"
                >
                  Bakke - Solheim
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Kontonummer
                </dt>
                <dd
                  className="typo-element"
                >
                  <span
                    className="c6"
                  >
                    <span>
                      3517
                    </span>
                    <span>
                      91
                    </span>
                    <span>
                      78779
                    </span>
                  </span>
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Inntekstsperiode
                </dt>
                <dd
                  className="typo-element"
                >
                  Månedlig
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Inntekt for perioden
                </dt>
                <dd
                  className="typo-element"
                >
                  40,062.00 NOK
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Refusjonstype
                </dt>
                <dd
                  className="typo-element"
                >
                  Ikke refusjon
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Refusjon til dato
                </dt>
                <dd
                  className="typo-element"
                >
                  11.02.2019
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Sykepenger fra og med
                </dt>
                <dd
                  className="typo-element"
                >
                  10.02.2019
                </dd>
              </div>
            </dl>
          </li>
        </ol>
        <button
          className="knapp c7 knapp--hoved"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Vis alle arbeidsforhold
        </button>
      </article>
    </div>
    <ol>
      <li
        className="c8"
      >
        <div
          className="c9"
        >
          <h3
            className="typo-element"
          >
            Periode 1 - 29.04.2017
          </h3>
        </div>
        <article
          className="c10"
        >
          <article
            aria-labelledby="Helt tilfeldig ID"
            className="c3"
          >
            <h3
              className="typo-ingress"
              id="Helt tilfeldig ID"
            >
              Om foreldrepengeperioden
            </h3>
            <dl
              className="c4"
            >
              <div>
                <dt
                  className="typo-normal"
                >
                  Arbeidsprosent mor
                </dt>
                <dd
                  className="typo-element"
                >
                  58%
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Mors situasjon
                </dt>
                <dd
                  className="typo-element"
                >
                  nobis sit quo et quis
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Disponibel gradering
                </dt>
                <dd
                  className="typo-element"
                >
                  9%
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Forskyvelsesperiode
                </dt>
                <dd
                  className="typo-element"
                >
                  08.10.2014 - 22.08.2014
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Andre forskyvelsesperiode
                </dt>
                <dd
                  className="typo-element"
                >
                  09.12.2014 - 26.08.2014
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Midlertidig stans
                </dt>
                <dd
                  className="typo-element"
                >
                  2019-02-10
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Stansårsak
                </dt>
                <dd
                  className="typo-element"
                >
                  Avsluttet
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Avslått
                </dt>
                <dd
                  className="typo-element"
                >
                  Avslått
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Mødrekvote
                </dt>
                <dd
                  className="typo-element"
                >
                  —
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Rett til Mødrekvote
                </dt>
                <dd
                  className="typo-element"
                >
                  Ingen rett til mødrekvote
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Aleneomsorg Mor
                </dt>
                <dd
                  className="typo-element"
                >
                  —
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Fedrekvote
                </dt>
                <dd
                  className="typo-element"
                >
                  —
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Rett til Fedrekvote
                </dt>
                <dd
                  className="typo-element"
                >
                  Rett til fedrekvote
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Aleneomsorg Far
                </dt>
                <dd
                  className="typo-element"
                >
                  —
                </dd>
              </div>
            </dl>
          </article>
          <section
            className="c11"
          >
            <h4
              className="typo-ingress"
            >
              Utbetalinger
            </h4>
            <div
              className="c12 typo-normal"
            >
              <table
                role="table"
              >
                <thead
                  role="rowgroup"
                >
                  <tr
                    role="row"
                  >
                    <th
                      role="columnheader"
                    >
                      Reg.dato
                    </th>
                    <th
                      role="columnheader"
                    >
                      Type
                    </th>
                    <th
                      role="columnheader"
                    >
                      Periode
                    </th>
                    <th
                      role="columnheader"
                    >
                      Utb.grad
                    </th>
                    <th
                      role="columnheader"
                    >
                      Dagsats
                    </th>
                    <th
                      role="columnheader"
                    >
                      Bruttobeløp
                    </th>
                    <th
                      role="columnheader"
                    >
                      Arb.giver
                    </th>
                    <th
                      role="columnheader"
                    >
                      Org.nummer
                    </th>
                  </tr>
                </thead>
                <tbody
                  role="rowgroup"
                >
                  <tr
                    role="row"
                  >
                    <td
                      role="cell"
                    >
                      25.06.2018
                    </td>
                    <td
                      role="cell"
                    >
                      KONTOØVERFØRING
                    </td>
                    <td
                      role="cell"
                    >
                      14.01.2018 - 15.07.2017
                    </td>
                    <td
                      role="cell"
                    >
                      79%
                    </td>
                    <td
                      role="cell"
                    >
                      705.00 NOK
                    </td>
                    <td
                      role="cell"
                    >
                      466.00 NOK
                    </td>
                    <td
                      role="cell"
                    >
                      Vedvik - Lunde
                    </td>
                    <td
                      role="cell"
                    >
                      1234567890
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              className="alertstripe alertstripe--info"
            >
              <span
                aria-label="info"
                className="alertstripe__ikon"
              >
                <svg
                  focusable="false"
                  height="1.5em"
                  kind="info-sirkel-fyll"
                  viewBox="0 0 24 24"
                  width="1.5em"
                >
                  <g
                    fill="none"
                  >
                    <path
                      d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                      fill="#669DB4"
                    />
                    <path
                      d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                      fill="#FFF"
                    />
                  </g>
                </svg>
              </span>
              <div
                className="typo-normal alertstripe__tekst"
              >
                For detaljer om utførte utbetalinger 
                <a
                  href="/person/10108000398/utbetaling"
                  onClick={[Function]}
                >
                  se Utbetalingsoversikten.
                </a>
              </div>
            </div>
          </section>
        </article>
      </li>
    </ol>
  </div>
</article>
`;
